packages/std-client/src/getBurnerWallet.ts

```diff
@@ -0,0 +1,38 @@
+import { Wallet } from "ethers";
+import { BehaviorSubject } from "rxjs";
+
+export function getBurnerWallet(cacheKey = "mud:burnerWallet"): BehaviorSubject<string> {
+  const cachedPrivateKey = localStorage.getItem(cacheKey);
+  const subject = cachedPrivateKey
+    ? new BehaviorSubject(cachedPrivateKey)
+    : (() => {
+        // TODO: move to viem wallet
+        const wallet = Wallet.createRandom();
+        console.log("New burner wallet created:", wallet.address);
+        localStorage.setItem(cacheKey, wallet.privateKey);
+        return new BehaviorSubject(wallet.privateKey);
+      })();
+
+  window.addEventListener("storage", function listener(event) {
+    // Clean up
+    if (subject.closed) {
+      window.removeEventListener("storage", listener);
+      return;
+    }
+
+    if (event.key !== cacheKey) return;
+    if (event.storageArea !== localStorage) return;
+
+    if (!event.newValue) {
+      // We'll intentionally not create a new burner wallet here to avoid potential infinite
+      // loop issues, and just warn the user. A refresh will go through the logic above to
+      // create a new burner wallet.
+      console.warn("Burner wallet removed from cache! You may need to reload to create a new wallet.");
+      return;
+    }
+
+    subject.next(event.newValue);
+  });
+
+  return subject;
```

Comment:
I've seen in some of our code that we return this using `asObservable`. Under what conditions does this make sense over returning the subject itself? I assume it would make it somewhat "read only" (but then we'd lose out on the `.get()` for the current value)?

Reply:
That's exactly right, `asObservable` makes it readonly but loses some of the `Subject` functionality. In this case `.get()` is pretty important so makes sense to not turn it into an observable.